tests/test_nblist.py

```diff
@@ -376,7 +380,114 @@ def test_nblist_max_interactions(block_size, tiles):
     nblist = custom_ops.Neighborlist_f32(coords.shape[0])
     max_ixn_count = nblist.get_max_ixn_count()
     test_ixn_list = nblist.get_nblist(coords, box, cutoff)
+    assert compute_mean_block_occupancy(nblist, coords, box, cutoff) == 1.0
     assert len(test_ixn_list) == tiles
     for tile_ixns in test_ixn_list:
         assert len(tile_ixns) > 0
     assert nblist.get_tile_ixn_count() * block_size == max_ixn_count
+
+
+@pytest.fixture(scope="module", params=["solvent", "complex"])
+def hif2a_single_topology_leg(request):
+    return setup_hif2a_initial_state(request.param)
+
+
+def setup_hif2a_initial_state(host_name: str):
+    forcefield = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+    host_config: Optional[HostConfig] = None
+
+    mol_a, mol_b, core = get_hif2a_ligand_pair_single_topology()
+    if host_name == "complex":
+        with resources.path("timemachine.testsystems.data", "hif2a_nowater_min.pdb") as protein_path:
+            host_sys, host_conf, box, _, num_water_atoms = build_protein_system(
+                str(protein_path), forcefield.protein_ff, forcefield.water_ff, mols=[mol_a, mol_b]
+            )
+            box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, num_water_atoms)
+    elif host_name == "solvent":
+        host_sys, host_conf, box, _ = build_water_system(4.0, forcefield.water_ff, mols=[mol_a, mol_b])
+        box += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes
+        host_config = HostConfig(host_sys, host_conf, box, host_conf.shape[0])
+    else:
+        assert 0, "Invalid host name"
+
+    st = SingleTopology(mol_a, mol_b, core, forcefield)
+    host = setup_optimized_host(st, host_config)
+
+    lambda_grid = np.array([0.0])
+
+    initial_state = setup_initial_states(st, host, DEFAULT_TEMP, lambda_grid, seed=2024, min_cutoff=None)[0]
+
+    return st, host, host_name, initial_state
+
+
+def compute_mean_block_occupancy(
+    nblist, frame: NDArray, box: NDArray, cutoff: float, column_block: int = 32, row_block: int = 32
+) -> float:
+    assert column_block == 32
+    assert column_block == row_block
+    N = frame.shape[0]
+    assert nblist.get_num_row_idxs() == N
+    ixn_list = nblist.get_nblist(frame, box, cutoff)
+    block_occupancies = []
+    max_ixn_per_block = column_block * row_block * ((N + row_block - 1) // row_block)
```

Comment:
why the extra  `((N + row_block - 1) // row_block)`?

Reply:
Because the structure of `ixn_list` from `nblist.get_nblist` returns the full rectangular block of row atoms by all column atoms. So it is the number of ixns in a single tile times the number of tiles.